### PR TITLE
[Research] Latent Perceptual Loss (LPL) for Stable Diffusion XL

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?

An implementation of Latent Perceptual Loss (LPL) for training Stable Diffusion XL models, based on the paper "Boosting Latent Diffusion with Perceptual Objectives" (Berrada et al., 2025). LPL is a perceptual loss that operates in the latent space of a VAE, helping to improve the quality and consistency of generated images by bridging the disconnect between the diffusion model and the autoencoder decoder.
 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sayakpaul — COMMENTED** [review]
Thank you! Very clean stuff!

Should we

1. Credit the first author for helping with a reference implementation? 
2. Should we apply LoRA on `unet` instead of full fine-tuning?
3. Change `lpl_sdxl.py` to `train_sdxl_lpl.py`?

Maybe update the PR description with a visual example from full fine-tuning run?

File: examples/research_projects/lpl/README.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    --lpl_norm_type shared \             # Normalization type: "default", "shared" (recommended), or "batch"
+    --lpl_pow_law \                      # Use power law weighting for layers
+    --lpl_num_blocks 4 \                 # Number of up blocks to use (1-4)
+    --lpl_remove_outliers \              # Remove outliers in feature maps
+    --lpl_scale \                        # Scale LPL loss by noise level weights
+    --lpl_start 0 \                      # Step to start applying LPL
+    # ... other training arguments ...
+```
+
+### Key Parameters
```
Comment (sayakpaul) [inline examples/research_projects/lpl/README.md:66]: Should also include a full representative training command. Currently, we only show LPL-specific bits in a command but lack one where a full-blown training can be launched (dataset_name, batch_size, etc.).
